### ISSUE TITLE
Error message when opening terminal

### ISSUE THREAD

**Issue by FabioAntunes** [fish-shell/fish-shell#2273]
This is what I'm getting sometimes, when I open up the terminal:
`Unable to make or open a FIFO for universal variables with path '/run/user/1000/fishd.74d02bd5b371.notifier': File exists`

Coud anyone explain why this happens? Since it's only sometimes and after doing a google search I couldn't find anything..

Some info relevant info, or not:
Fish version:  2.2.0
Distro: elementary OS
Terminal:  Pantheon

**Comment by faho** [comment]
> Terminal: Pantheon

I think the terminal is called Pantheon-Terminal, Pantheon is the DE. Are you using that one? Shouldn't make much of a difference here, though.

> Distro: elementary OS

For the others: This is an Ubuntu-derivative - they have their own repos for the DE and such, but use Ubuntu repos for everything else IIRC.

@FabioAntunes: Do you use sudo or su or pkexec or something? What's your "XDG_RUNTIME_DIR" (`echo $XDG_RUNTIME_DIR`)?

Can you narrow down when this happens? After you've sudo'd? Can you post the output of `getfacl /run/user/1000/fishd.*.notifier` after it happens?

Does fish go on running? (Presumably yes, but without access to universal variables)

This error is reported in env_universal_common.cpp's "make_pipe", but I don't quite understand the comments:

```
            // We explicitly do NOT report an error for ENOENT or EACCESS
            // This works around #1955, where $XDG_RUNTIME_DIR may get a bogus value under suc
            if (err != ENOENT && err != EPERM)
            {
                report_error(err, L"Unable to make or open a FIFO for universal variables with path '%ls'", vars_path.c_str());
            }
```

Why does the comment say ENOENT and EACCESS when the code checks for ENOENT and EPERM?

**Comment by ridiculousfish** [comment]
Well that was sloppy of me. I think the code is correct and the comment is wrong. @FabioAntunes can you check the permissions of `/run/user/1000/fishd.74d02bd5b371.notifier`? Is it owned by root? It might be another manifestation of https://github.com/fish-shell/fish-shell/issues/1955 .

**Comment by FabioAntunes** [comment]
So running `getfacl /run/user/1000/fishd.*.notifier`

````
getfacl: Removing leading '/' from absolute path names
# file: run/user/1000/fishd.74d02bd5b371.notifier
# owner: fabio
# group: fabio
user::rw-
group::---
other::---
````

@faho I can't tell you what are the exact occasions this occurs, it's just sometimes I can't or I don't remember what happened before to cause this.

Also I don't think fish carries on, at least on the tab where I get the error, I usually open a new tab and close the one where the error pops up.

One thing I forgot to mention and don't know if it's important. I have oh-my-fish installed, the latest version.

**Comment by jrobeson** [comment]
when my X session starts I end up wit two terminals, one always has this error, and the other one always doesn't. race condition maybe?

**Comment by ihipop** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=753882#c17

`vim /etc/pam.d/su` , add `session required pam_loginuid.so` before any `include common-session`
then use `su` with `-l` option, eg:
```sudo su -l```

**Comment by krader1961** [comment]
@FabioAntunes Ping? Is this still an issue for you? Did you try @ihipop's solution?

**Comment by FabioAntunes** [comment]
@krader1961 To be honest it's been a while since I got this and the biggest problem is the randomness of this issue making it harder to replicate.

When I reported this issues I was using fish 2.2 and omf,now using fish 2.3 with fisherman and I haven't got this error message since I'm using this setup. I think it's safe to close this issue for now.

**Comment by krader1961** [comment]
From my comment in issue #3426:

> This is just a really unlikely, but not impossible, race condition. Look at the universal_notifier_named_pipe_t::make_pipe() method. It attempts to open the named pipe. If that fails it attempts to create the named pipe then repeat the open attempt. If you launch two fish instances simultaneously on a system with no previously running fish programs then it is possible for both to fail the initial open and one of them to fail the subsequent mkfifo() call because the other has already created the named pipe.

**Comment by jrobeson** [comment]
it's not unlikely. I get it almost every boot. as mentioned in https://github.com/fish-shell/fish-shell/issues/2273#issuecomment-133636329

**Comment by krader1961** [comment]
This has been fixed by commit 8ba2a4c.
